File: shared/keycloak-initializer-core/src/main/java/com/visor/school/keycloak/detection/InitializerStateEvaluator.java
package com.visor.school.keycloak.detection;

import com.visor.school.keycloak.model.InitializationDecision;
import com.visor.school.keycloak.model.KeycloakBlueprint;

/**
 * Determines whether the Keycloak initializer should apply the provided blueprint.
 */
@FunctionalInterface
public interface InitializerStateEvaluator {
    InitializationDecision evaluate(KeycloakBlueprint blueprint);
}

File: shared/keycloak-initializer-core/src/main/kotlin/com/visor/school/keycloak/detection/InitializerStateEvaluator.kt
package com.visor.school.keycloak.detection

import com.visor.school.keycloak.model.InitializationDecision
import com.visor.school.keycloak.model.KeycloakBlueprint

/**
 * Determines whether the Keycloak initializer should apply the provided blueprint.
 */
fun interface InitializerStateEvaluator {
    fun evaluate(blueprint: KeycloakBlueprint): InitializationDecision
}
